magento/pwa-studio
# PWA-602: Add to Shopping Bag from PDP

## Description

Adding an item to the shopping bag should add the item and update the cart trigger.
If there is a problem, it should show a message in the form. If you're on `develop` mode then there will be an error logged to console.

## Related Issue





Closes [PWA-602](https://jira.corp.magento.com/browse/PWA-602).

## Acceptance 



### Verification Stakeholders

### Specification



### Verification Steps


1. Go to the PDP and add an item. It should work.
2. Add an item that is out of stock. It should show an error about quantity.
3. Add an item but trigger an unknown error such as an invalid user token. It should render a form level error.

## Screenshots / Screen Captures (if appropriate)

**Successful Add**
[![Image from Gyazo](https://i.gyazo.com/ed2c91346a847ffba5f1d2aa622eb0fc.gif)](https://gyazo.com/ed2c91346a847ffba5f1d2aa622eb0fc)

**Quantity Error**
[![Image from Gyazo](https://i.gyazo.com/12841847b303aa93cb528bc1a2f419d8.gif)](https://gyazo.com/12841847b303aa93cb528bc1a2f419d8)

**Unknown Error**
[![Image from Gyazo](https://i.gyazo.com/bbb192882b1824d05d1eb7fbfa6b4b60.gif)](https://gyazo.com/bbb192882b1824d05d1eb7fbfa6b4b60)

## Checklist

* I have updated the documentation accordingly, if necessary.
